tests/scenarios/MiddlewareTestA/MiddlewareA.cpp

```diff
@@ -0,0 +1,209 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include <test_config.h>
+#include <appmodel.h>
+#include <algorithm>
+#include <ShlObj.h>
+#include <filesystem>
+#include <dependency_helper.h>
+
+using namespace std::literals;
+
+#define Registry_Key_1                 "Software\\random_key_2\\path\\latest"
+#define Registry_Key_2(v)              "Software\\random_key\\abc\\" + v
+
+#define Registry_Value_Version         "Version"
+#define Registry_Value_Full_Path       "FullPath"
+
+#define Environment_Variable_1         "LIB_PATH"
+#define Environment_Variable_2         "PKG_VER"
+
+int main(int argc, const char** argv)
+{
+    auto result = parse_args(argc, argv);
+
+    test_initialize("MiddlewareW Tests", 4);
+
+    if (auto vclib_package = query_package_with_name(VCLibsName))
+    {
+        {
+            test_begin("Middleware test RegQueryValue able to fetch HKLM data created by the fixup with proper replacements");
+
+            HKEY res;
+            if (RegOpenKeyExA(HKEY_LOCAL_MACHINE, Registry_Key_1, 0, KEY_ALL_ACCESS, &res) == ERROR_SUCCESS)

```

Comment:
Please use correct WIL MACRO's for status comparison.

Reply:
Which one? It returns a LSTATUS not HRESULT